cranelift/codegen/src/isa/x64/inst/mod.rs

```diff
@@ -211,6 +226,38 @@ impl Inst {
         Inst::Mov_R_R { is_64, src, dst }
     }
 
+    // TO_ASK: Should be combined with the mov_r_r via some if-statements to handle return?
```

Comment:
Not 100% sure I understand the question, but .. let's keep the XMM and integer insns separate.  So I think the answer is then "no".

Reply:
What prompted this comment was here in lower.rs:

https://github.com/bytecodealliance/wasmtime/blob/48521393ae6b6e014902fac8fe70ac2c1b221880/cranelift/codegen/src/isa/x64/lower.rs#L187

For a lot of opcodes we can specify from here which Inst::mov variant we want to call but for return for example, multiple types apply and I found I was going to only be calling one version which might not be the correct one. I was going to check types in lower.rs, but instead I created a mov specific to the return op and put the decision logic there.

Reply:
Ah, that is a good point.  Yes.  True.  Nevertheless, the general flavour of the rest of the instruction selector (lower.rs) is that we keep int and vec instructions separate, and we don't hand that problem to the instruction-representation layer.  So the lowering logic to handle `Opcode::FallthroughReturn | Opcode::Return` will need to do the type test itself, rather than handing it off to `return_mov_r_r`.  FWIW I think there are only very few places like this, where we need to check int-vs-vec/float types at the lowering level -- in prologue construction is the only other place I can think of.